### DISCUSSION TITLE
How to install metallb on k3s cluster? how to include kubeproxy with strictarp parameter?

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [k3s-io/k3s#7124] [General]
How to install metallb on k3s cluster? how to include kubeproxy with strictarp parameter?

**Comment by brandond** [comment]
Install it using the instructions on the metallb project site.

Kube-proxy args can be set with the --kube-proxy-arg CLI flag.

**Reply by martinenkoEduard** [reply]
Where should I set it if I use official install script? Should I change systemd service file?

**Reply by brandond** [reply]
you can do it there, or use the config file. its up to you which you prefer.

**Reply by martinenkoEduard** [reply]
How do I set it in config file?

**Reply by brandond** [reply]
https://docs.k3s.io/installation/configuration#configuration-file

**Reply by HWiese1980** [reply]
I've been searching for how to specifically set the strictarp parameter using --kube-proxy-arg, but couldn't find any example. Could you please add an example of how to do this in particular when launching k3s? 

`k3s server [other params] --kube-proxy-arg ???`

That would be awesome!

**Reply by brandond** [reply]
It works the same as all the other component arg pass-through flags:

```
--kube-proxy-arg=ipvs-strict-arp=true
```

or in the config file:

```yaml
kube-proxy-arg:
  - ipvs-strict-arp=true
```

See https://kubernetes.io/docs/reference/command-line-tools-reference/kube-proxy/ for available args.

**Reply by HWiese1980** [reply]
Very cool! Thank you!
